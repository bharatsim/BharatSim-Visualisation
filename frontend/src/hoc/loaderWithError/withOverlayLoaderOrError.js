import React, { useState } from 'react';
import { OverlayLoaderOrErrorContextProvider } from '../../contexts/overlayLoaderOrErrorContext';
import OverlayLoader from '../../uiComponent/OverlayLoader';
import OverlayError from '../../uiComponent/OverlayError';
import {useDispatch, useSelector} from "react-redux";
import * as actions from "./actions";

function withOverlayLoaderOrError(WrappedComponent) {
  return (props) => {
    const loaderCount = useSelector((state)=> state.loaderOrError.loaderCount)
    const isError =  useSelector((state)=> state.loaderOrError.isError)
    const errorConfig =  useSelector((state)=> state.loaderOrError.errorConfig)

    const dispatch =  useDispatch()
    function startLoader() {
      dispatch(actions.startLoader())
    }

    function stopLoader() {
      dispatch(actions.stopLoader())
    }

    function showError(errorConfigs) {
      dispatch(actions.showErrors(errorConfigs))
    }

    function hideError() {
      dispatch(actions.hideErrors())
    }
    return (
      <OverlayLoaderOrErrorContextProvider value={{ stopLoader, startLoader, showError }}>
        <WrappedComponent {...props} />
        {loaderCount >= 1 && <OverlayLoader isLoading={loaderCount >= 1} />}
        {isError && (
          <OverlayError
            isError={isError}
            errorConfig={errorConfig}
            hideError={hideError}
          />
        )}
      </OverlayLoaderOrErrorContextProvider>
    );
  };
}

export default withOverlayLoaderOrError;
