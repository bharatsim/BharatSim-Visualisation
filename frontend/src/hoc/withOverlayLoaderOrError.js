import React, { useState } from 'react';
import { OverlayLoaderContextProvider } from '../contexts/overlayLoaderContext';
import OverlayLoader from '../uiComponent/OverlayLoader';
import OverlayError from '../uiComponent/OverlayError';
import { errorTypes } from '../constants/loaderAndErrorMessages';

function withOverlayLoaderOrError(WrappedComponent) {
  return (props) => {
    const [loader, setLoader] = useState({ loadersCount: 0 });
    const [error, setError] = useState({
      isError: false,
      errorCode: errorTypes.DEFAULT,
      data: undefined,
      query: undefined,
    });

    function startLoader() {
      setLoader((prevState) => ({
        loadersCount: prevState.loadersCount + 1,
      }));
    }

    function stopLoader() {
      setLoader((prevState) => {
        if (prevState.loadersCount === 0) {
          return { loadersCount: 0 };
        }
        return { loadersCount: prevState.loadersCount - 1 };
      });
    }

    function stopAllLoaders() {
      setLoader({ loadersCount: 0, message: '' });
    }

    function showError(errorCode) {
      stopAllLoaders();
      setError({ isError: true, errorCode });
    }

    function hideError() {
      setError({ isError: false, errorCode: errorTypes.DEFAULT });
    }

    return (
      <OverlayLoaderContextProvider value={{ stopLoader, startLoader, showError }}>
        <WrappedComponent {...props} />
        {loader.loadersCount >= 1 && <OverlayLoader isLoading={loader.loadersCount >= 1} />}
        {error.isError && (
          <OverlayError
            isError={error.isError}
            errorCode={error.errorCode}
            onClose={hideError}
            data={error.data}
          />
        )}
      </OverlayLoaderContextProvider>
    );
  };
}

export default withOverlayLoaderOrError;
