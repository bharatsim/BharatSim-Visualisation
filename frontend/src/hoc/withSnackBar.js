import React from 'react';
import { SnackbarProvider } from 'notistack';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import failureIcon from '../assets/images/error-icon.svg';
import successIcon from '../assets/images/success-icon.svg';

const useSnackBarStyles = makeStyles((theme) => ({
  contentRoot: {
    padding: theme.spacing(1, 4),
  },
  message: {
    padding: 0,
  },
  variantSuccess: {
    backgroundColor: `#EAF4E5 !important`,
    borderColor: `${theme.palette.success.main}33`,
    border: '1px solid',
    color: `${theme.palette.success.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
  variantError: {
    backgroundColor: `#FBE9E4 !important`,
    borderColor: `${theme.palette.error.light}33`,
    border: '1px solid',
    color: `${theme.palette.error.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
}));

const useIconStyles = makeStyles((theme) => ({
  iconRoot: {
    marginRight: theme.spacing(2),
    height: theme.spacing(10),
    width: theme.spacing(10),
  },
}));

function withSnackBar(WrappedComponent) {
  return (props) => {
    const snackBarClasses = useSnackBarStyles();
    const iconClasses = useIconStyles();
    const snackBarRef = React.createRef();
    const onClickDismiss = (key) => () => {
      snackBarRef.current.closeSnackbar(key);
    };
    return (
      <SnackbarProvider
        maxSnack={3}
        ref={snackBarRef}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        classes={snackBarClasses}
        iconVariant={{
          error: <img src={failureIcon} alt="error" className={iconClasses.iconRoot} />,
          success: <img src={successIcon} alt="success" className={iconClasses.iconRoot} />,
        }}
        action={(key) => <Button onClick={onClickDismiss(key)}>Dismiss</Button>}
      >
        <WrappedComponent {...props} />
      </SnackbarProvider>
    );
  };
}

export default withSnackBar;
