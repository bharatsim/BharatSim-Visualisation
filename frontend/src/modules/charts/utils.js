import { chartStyleConfig, getColor } from './chartStyleConfig';

function trasformDataForChart(csvData, xColumn, yColumns) {
  return {
    labels: csvData && csvData.data[xColumn],
    datasets:
      csvData &&
      yColumns.map((yColumn, index) => {
        return {
          ...chartStyleConfig,
          label: yColumn,
          borderColor: getColor(index),
          backgroundColor: getColor(index),
          data: csvData && csvData.data[yColumn],
        };
      }),
  };
}

function getYaxisNames(yColumns) {
  return yColumns.map((yColumn) => yColumn.name);
}

export { trasformDataForChart, getYaxisNames };
