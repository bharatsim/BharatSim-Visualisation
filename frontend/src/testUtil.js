// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
