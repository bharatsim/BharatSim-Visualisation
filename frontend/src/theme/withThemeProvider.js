import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
