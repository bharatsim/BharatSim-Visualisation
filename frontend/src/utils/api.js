import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  addNewDashboard: async ({ name, projectId }) => {
    return uploadData({
      url: serviceURL.INSERT_DASHBOARD,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets: [], layout: [], name, count: 0, projectId },
      }),
      isCustomErrorHandler: true,
    });
  },

  getAllDashBoardByProjectId: async (projectId) =>
    fetchData({
      url: serviceURL.DASHBOARD_URL,
      query: { projectId, columns: ['name', '_id'] },
    }),

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema, dashboardId }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
        { name: 'dashboardId', value: dashboardId },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async (dashboardId) =>
    fetchData({
      url: serviceURL.DATA_SOURCES,
      query: { dashboardId },
    }),

  getData: async (datasource, columns) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query: { columns } }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      isCustomErrorHandler: true,
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
