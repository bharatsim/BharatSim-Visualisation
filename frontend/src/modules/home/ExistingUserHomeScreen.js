import React from 'react';
import PropTypes from 'prop-types';

import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Button from '@material-ui/core/Button';
import { useHistory } from 'react-router-dom';
import ProjectMetadataCard from './ProjectMetadataCard';
import theme from '../../theme/theme';
import TabPanel from '../../uiComponent/TabPanel';
import ButtonGroup from '../../uiComponent/ButtonGroup';

const styles = makeStyles(() => ({
  projectListContainer: {
    marginTop: theme.spacing(11),
  },
}));

function ExistingUserHomeScreen({ recentProjects }) {
  const [selectedTab] = React.useState(0);
  const history = useHistory();

  const classes = styles();

  function openProject(id) {
    history.push(`/projects/${id}/configure-dataset`);
  }

  function createNewProject() {
    history.push('/projects/create');
  }

  return (
    <Box>
      <Box display="flex" justifyContent="space-between">
        <Tabs value={selectedTab} indicatorColor="primary" aria-label="disabled tabs example">
          <Tab label="Recent Projects" />
        </Tabs>
        <ButtonGroup>
          <Button variant="contained" color="primary" size="small" onClick={createNewProject}>
            Add New
          </Button>
        </ButtonGroup>
      </Box>

      <Box className={classes.projectListContainer}>
        <TabPanel value={selectedTab} index={0}>
          {recentProjects && (
            <Grid container spacing={8} xl={12}>
              {recentProjects.map((project) => {
                const { _id } = project;
                return (
                  <Grid item xs={3} key={_id}>
                    <ProjectMetadataCard project={project} onProjectClick={openProject} />
                  </Grid>
                );
              })}
            </Grid>
          )}
        </TabPanel>
      </Box>
    </Box>
  );
}

ExistingUserHomeScreen.propTypes = {
  recentProjects: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      _id: PropTypes.string,
    }),
  ).isRequired,
};

export default ExistingUserHomeScreen;
