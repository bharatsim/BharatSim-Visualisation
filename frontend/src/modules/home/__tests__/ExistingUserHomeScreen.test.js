import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderWithRedux as render } from '../../../testUtil';
import ExistingUserHomeScreen from '../ExistingUserHomeScreen';
import withSnackBar from '../../../hoc/snackbar/withSnackBar';
import { api } from '../../../utils/api';
import withThemeProvider from '../../../theme/withThemeProvider';
import withOverlayLoaderOrError from '../../../hoc/loaderWithError/withOverlayLoaderOrError';

jest.mock('../../../utils/api', () => ({
  api: {
    deleteProject: jest.fn().mockResolvedValue({
      deleted: 1,
    }),
    getDatasourcesForProject: jest.fn().mockResolvedValue({
      dataSources: [{ _id: 'datasourceId' }],
    }),
    getAllDatasources: jest.fn().mockResolvedValue({
      dataSources: [{ _id: 'datasourceId', name: 'fileName' }],
    }),
    deleteDatasource: jest.fn().mockResolvedValue({ deleted: 1 }),
  },
}));
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('Existing User Home Screen', () => {
  const ExistingUserHomeScreenWithProviders = withOverlayLoaderOrError(
    withThemeProvider(withSnackBar(ExistingUserHomeScreen)),
  );
  it('should match snapshot for given projects', () => {
    const { container } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: '1', name: 'project1' }]}
        setRecentProjects={jest.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
  it('should navigate to datesets library on click of datasets tab', async () => {
    const { getByText, findByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: '1', name: 'project1' }]}
        setRecentProjects={jest.fn()}
      />,
    );
    const datasetsTabButton = getByText('Datasets');
    fireEvent.click(datasetsTabButton);
    await findByText('fileName');
    expect(getByText('fileName')).toBeInTheDocument();
  });
  it('should navigate to project on click of project card', () => {
    const { getByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: '1', name: 'project1' }]}
        setRecentProjects={jest.fn()}
      />,
    );

    fireEvent.click(getByText('project1'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/projects/1/dashboard');
  });
  it('should create new project on click of add new button', () => {
    const { getByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: '1', name: 'project1' }]}
        setRecentProjects={jest.fn()}
      />,
    );

    fireEvent.click(getByText('Add New'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/projects/create');
  });
  // TODO: Update test after delete datasource implemented
  it('should delete project on click of delete option', async () => {
    const mockSetRecentProject = jest.fn();
    const { getByTestId, findByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: 'projectId', name: 'projectName' }]}
        setRecentProjects={mockSetRecentProject}
      />,
    );
    const projectMenuButton = getByTestId('project-menu');
    fireEvent.click(projectMenuButton);

    const deleteOptionForProject1 = getByTestId('delete-project-projectId');
    fireEvent.click(deleteOptionForProject1);

    const deleteProjectConfirmationButton = getByTestId('delete-project-confirmation-button');
    fireEvent.click(deleteProjectConfirmationButton);

    await findByText('Project projectName deleted successfully');

    expect(mockSetRecentProject).toHaveBeenCalledWith([]);
    expect(api.deleteProject).toHaveBeenCalledWith('projectId');
  });
  it('should show error if any while deleting project', async () => {
    const mockSetRecentProject = jest.fn();
    api.deleteProject.mockRejectedValueOnce();
    const { getByTestId, findByText, getByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: 'projectId', name: 'projectName' }]}
        setRecentProjects={mockSetRecentProject}
      />,
    );
    const projectMenuButton = getByTestId('project-menu');
    fireEvent.click(projectMenuButton);

    const deleteOptionForProject1 = getByTestId('delete-project-projectId');
    fireEvent.click(deleteOptionForProject1);

    const deleteProjectConfirmationButton = getByTestId('delete-project-confirmation-button');
    fireEvent.click(deleteProjectConfirmationButton);

    await findByText('Aw Snap! Failed to delete project projectName');

    expect(getByText('Aw Snap! Failed to delete project projectName')).toBeInTheDocument();
  });
  it('should show error if any while deleting datasources for project', async () => {
    const mockSetRecentProject = jest.fn();
    api.deleteDatasource.mockRejectedValueOnce();
    const { getByTestId, findByText, getByText } = render(
      <ExistingUserHomeScreenWithProviders
        recentProjects={[{ _id: 'projectId', name: 'projectName' }]}
        setRecentProjects={mockSetRecentProject}
      />,
    );
    const projectMenuButton = getByTestId('project-menu');
    fireEvent.click(projectMenuButton);

    const deleteOptionForProject1 = getByTestId('delete-project-projectId');
    fireEvent.click(deleteOptionForProject1);

    const deleteProjectConfirmationButton = getByTestId('delete-project-confirmation-button');
    fireEvent.click(deleteProjectConfirmationButton);

    await findByText('Aw Snap! Failed to delete datasource file for project projectName');

    expect(
      getByText('Aw Snap! Failed to delete datasource file for project projectName'),
    ).toBeInTheDocument();
  });
  it(
    'should not delete the datasources associated with project id ' +
      'on click of no from confirmation modal',
    async () => {
      const mockSetRecentProject = jest.fn();
      const { getByTestId, findByText, getByText } = render(
        <ExistingUserHomeScreenWithProviders
          recentProjects={[{ _id: 'projectId', name: 'projectName' }]}
          setRecentProjects={mockSetRecentProject}
        />,
      );
      const projectMenuButton = getByTestId('project-menu');
      fireEvent.click(projectMenuButton);

      const deleteOptionForProject1 = getByTestId('delete-project-projectId');
      fireEvent.click(deleteOptionForProject1);

      const NoRadioButton = getByTestId('no-radio-button');
      fireEvent.click(NoRadioButton);

      const deleteProjectConfirmationButton = getByTestId('delete-project-confirmation-button');
      fireEvent.click(deleteProjectConfirmationButton);

      await findByText('Project projectName deleted successfully');

      expect(getByText('Project projectName deleted successfully')).toBeInTheDocument();
    },
  );
  it(
    'should not delete the datasources associated with project' +
      ' id if datasources are not present',
    async () => {
      const mockSetRecentProject = jest.fn();
      api.getDatasourcesForProject.mockResolvedValue({
        dataSources: [],
      });
      const { getByTestId, findByText, getByText } = render(
        <ExistingUserHomeScreenWithProviders
          recentProjects={[{ _id: 'projectId', name: 'projectName' }]}
          setRecentProjects={mockSetRecentProject}
        />,
      );
      const projectMenuButton = getByTestId('project-menu');
      fireEvent.click(projectMenuButton);

      const deleteOptionForProject1 = getByTestId('delete-project-projectId');
      fireEvent.click(deleteOptionForProject1);

      const deleteProjectConfirmationButton = getByTestId('delete-project-confirmation-button');
      fireEvent.click(deleteProjectConfirmationButton);

      await findByText('Project projectName deleted successfully');

      expect(getByText('Project projectName deleted successfully')).toBeInTheDocument();
    },
  );
});
