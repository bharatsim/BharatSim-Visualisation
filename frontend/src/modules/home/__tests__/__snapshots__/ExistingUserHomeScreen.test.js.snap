// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Existing User Home Screen should match snapshot for given projects 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            aria-label="disabled tabs example"
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Recent Projects
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-4 PrivateTabIndicator-colorPrimary-5 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-buttonGroup-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add New
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-10 makeStyles-projectListContainer-1"
    >
      <div
        aria-labelledby="simple-tab-0"
        id="simple-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              class="MuiBox-root MuiBox-root-12 makeStyles-cardContainer-11"
            >
              <div
                class="MuiBox-root MuiBox-root-13"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  project1
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
